client/core/core.go

```diff
@@ -1688,15 +1711,19 @@ func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) err
 		}
 	}
 
-	// Check that the new password works. If the new password is empty, skip
-	// this step, since an empty password signifies an unencrypted wallet.
 	wasUnlocked := wallet.unlocked()
+	var encPW []byte
 	if newPasswordSet {
-		err = wallet.Wallet.Unlock(string(newPW))
+		encPW, err = crypter.Encrypt(newPW)
+		if err != nil {
+			return newError(encryptionErr, "encryption error: %v", err)
+		}
+		err = c.setWalletPassword(wallet, newPW, encPW)
 		if err != nil {
-			return newError(authErr, "Error unlocking wallet. Is the new password correct?: %v", err)
+			return err
 		}
 	}
+	// TODO: else wallet.encPW = nil?
```

Comment:
I believe you can move most of the `if newPasswordSet` blocks in `SetWalletPassword` and `ReconfigureWallet` into the unexported `setWalletPassword`, you'll just pass the unencrypted PW instead of the encrytped.

You're handling the `encPW = nil` part correctly in `setWalletPassword`.

Reply:
good point 👍

Reply:
I would need to pass the crypter also, as setWalletPassword is taking care
of encrypting the password if it's not empty.